Add ListenBrainz architecture docs (#1826)
metabrainz/listenbrainz-server: docs/maintainers/dumps.rst

Data Dumps
==========

Check FTP Dumps age script
^^^^^^^^^^^^^^^^^^^^^^^^^^

> Dumps may fail in production due to many reasons. We have a script to check the latest dump available on the FTP is

Reviewer: link to the script.

younger than a specified timeframe. If the latest dump is older, an email is sent to the maintainers. This email
is usually responsible for bringing dump failures to the notice of maintainers. This script is part of the ListenBrainz
cronjobs and is scheduled to run a few hours after the regular dump times. If dumps are not working but no email was
received by the maintainers, it is possible that the cron jobs are not setup properly.

Logs
^^^^
Looking at the logs is a good starting point to debug dump failures, the log file is located at :file:`/logs/dumps.log`
inside the listenbrainz-cron-prod container at the time of writing. You can check the crontab file for the current
location of the log file. You can open a bash shell in the container by doing something like
:code:`docker exec -it listenbrainz-cron-prod bash`.

> After that you can use :command:`tail` to view the latest logs. :command:`cat` is also available but the logs file will
> be huge so the experience using it will not be nice. For example: :code:`tail -n 500 /logs/dumps.log` will list the last
> 500 lines of the log file.

Reviewer: ```suggestion
This file is large, so use :command:`tail` instead of :command:`cat` to view the logs. For example: 
:code:`tail -n 500 /logs/dumps.log` will list the last 500 lines of the log file.
```

From the log file, you should probably be able to see whether the error occurred in python part of the code or bash

> script. If you see a python stack trace, it is likely that sentry recorded the error too. The sentry view sometimes

Reviewer: link to sentry

offers more details so searching sentry for this error can be helpful.

Manually triggering dumps
^^^^^^^^^^^^^^^^^^^^^^^^^

> Once you have debugged and fixed the issue, you'll probably want to trigger new dumps manually. A few things need to be

Reviewer: ```suggestion
If you want to re-run a dump after it fails, or manually trigger a dump then you can run the dump script manually. A few things need to be
```

kept in mind while doing this, the python command invoked to do the dump accepts a id to number the dump. If no id
specified, the script will look in the database for the last id, add 1 to it and use it for the dump.

> 

Reviewer: it'd be good to include the sql statement here - if someone is debugging something then it's nice to not have to think how to do somethign
